safenode/src/client/api.rs

```diff
@@ -76,16 +80,36 @@ impl Client {
                 }
 
                 info!("Client waiting for a network event");
-                let event = match network_event_receiver.recv().await {
-                    Some(event) => event,
-                    None => {
-                        error!("The `NetworkEvent` channel has been closed");
-                        continue;
+
+                tokio::select! {
```

Comment:
It's perhaps cleaner with `match tokio::time::timeout(INACTIVITY_TIMEOUT, network_event_receiver.recv()).await { ... }` ?